### ISSUE TITLE
ACE Editor Autocomplete to open on `.`

### ISSUE THREAD

**Issue by ktalebian** [ajaxorg/ace#3690]
If you enable live autocomplete, the tooltip opens up as you type things. However, it will never open up on pressing . until you type something else after.

So if I were to type fo there will be a tooltip trying to match fo to something (for or function, etc). However, if type fo. then the tooltip disappears, and it will not open until I type a further letter.

I have custom autocompletion keys, and I really want the tooltip to open up with all my suggestions.

Let's say I have user as a special key with name, age, profile as the suggestions. I want to be able to type event. and the tooltip to show name, age, profile. Right now, I have to type event.n for example to get name to show as a suggestion

**Comment by maxHoo** [comment]
I am in the same situation for a long time and made many posts, nut still get no answer yet. It seems except largely modify source code, cannot solve this problem.

**Comment by mkslanc** [comment]
Fixed by #5147 . To use this feature you just need to register completer with `triggerCharacters` similar to 
```javascript
 editor.completers = [
            {
                getCompletions: function (editor, session, pos, prefix, callback) {
                    var completions = [
                        {
                            caption: "append",
                            value: "append"
                        }, {
                            caption: "all",
                            value: "all"
                        }
                    ];
                    callback(null, completions);
                },
                triggerCharacters: ["."]
            }
        ];
```
